source/MaterialXGenShader/Util.cpp

```diff
@@ -608,4 +608,147 @@ void findRenderableElements(const DocumentPtr& doc, std::vector<TypedElementPtr>
     }
 }
 
+ValueElementPtr findNodeDefChild(const string& path, DocumentPtr doc, const string& target)
```

Comment:
Thought this one might be generally useful to find a nodedef elem based on a node elem.